classes/fullstack/project1/client/src/store/slices/auth.slice.ts

```diff
@@ -0,0 +1,28 @@
+import { PayloadAction, createSlice } from "@reduxjs/toolkit";
+import { AdminRegister, Role } from "../../types"
+
+type AuthState = {
+  admin: AdminRegister
+}
+
+const initialState: AuthState = {
+  admin: {
+    username: '',
+    firstName: '',
+    lastName: '',
+    role: Role.THREE,
```

Comment:
type here can be either Role | null its safer than defaulting to an exisiting permission set